### DISCUSSION TITLE
Error in reported chuck position

### DISCUSSION THREAD

**Discussion by tl1949** [kachurovskiy/nanoels#228] [General]
I have been running the H4 controller successfully for a couple of years but I now have a problem with screwcutting that I haven’t had before. I eventually located the problem was with the reporting of the chuck position by the H4. I changed the rotary encoder but the problem remains.
I manually turned the chuck 100 times measuring its recorded position on the H4 for each revolution. Over the 100 revolutions, the total recorded angle was just over 40 degrees less than it should have been but the discrepancy for each revolution was not constant and varied by different amounts ie. both greater or less than the average of 0.4 degrees variance for each revolution.
I have checked the output of the rotary encoder on an oscilloscope and can see no evidence of missing pulses. The waveform is a good square wave but the pulse width does vary within a few percent. The rotary encoder has 600 steps and the firmware is set to 600 steps. The output waveform is consistent at 4.7 volts.
I will welcome any suggestions on what else to check or what might be causing the H4 to miscount the angle of the chuck now - when it has previously been accurate.

**Comment by kotovasia5120** [comment]
maybe the problem is in esp

**Comment by kachurovskiy** [comment]
If you are using software version below 12, try updating to the latest h4.ino

If you are using version 12, try reducing `ENCODER_FILTER` to 1 or disabling encoder filter entirely in https://github.com/kachurovskiy/nanoels/blob/main/h4/h4.ino#L1596C2-L1596C20 by changing that line to call `pcnt_filter_disable` instead.

**Reply by tl1949** [reply]
Thanks, Maxim, your suggestion has worked ie. the updated software and changing ENCODER_FILTER to 1.
